src/models/subject.ts

```diff
@@ -0,0 +1,157 @@
+}
+
+export class Subject {
+    public constructor(data: SubjectData) {
+        this.data = data;
+        this.identifier = Symbol("verification-subject");
+    }
+
+    public identifier: symbol;
+    public data: SubjectData;
+}
+
+// the Verification model is emitted by the oe-verification-grid as a CustomEvent
+export class SubjectWrapper implements ISubjectWrapper {
+    public constructor(data: ISubjectWrapper) {
+        this.subject = data.subject;
+        this.decisions = data.decisions ?? new Map();
+        this.url = data.url;
+        this.tag = data.tag;
+
+        this.identifier = Symbol("decision");
+    }
+
+    // aka: context
+    // this is the native data model used by the host application
+    // or this could be the csv row
+    public identifier: symbol;
+    public subject: Subject;
+
+    public clientCached = false;
+    public serverCached = false;
+
+    // each row can only have one verification decision
+    // but can have multiple classification decisions
+    // verification decisions will be reflected in the oe-confirmed
+    // column, while each classification will get its own row
+    public decisions: Map<symbol, IDecision[]>;
+    public url: string;
+    public tag: string;
+
+    public get decisionModels(): IDecision[] {
+        return Array.from(this.decisions.values()).flat();
+    }
+
+    public get verification(): Verification | undefined {
+        return this.decisionModels.find((decision) => decision.type === "verification") as Verification;
+    }
+
+    public get classifications(): Classification[] {
+        return this.decisionModels.filter((decision) => decision.type === "classification") as Classification[];
+    }
+
+    public addDecision(newDecision: IDecision): void {
+        const decision = newDecision;
+
+        // if there are two decisions that are mutually exclusive, we want to
+        // remove them before applying the new decision
+        if (decision instanceof Verification) {
+            const existingVerification = this.verification;
+            if (existingVerification) {
+                this.removeDecision(existingVerification);
+            }
+        } else if (decision instanceof Classification) {
+            const exclusiveToTags = this.classifications.filter(
+                (classification) => classification.tag.text === decision.tag.text,
+            );
+
+            for (const classification of exclusiveToTags) {
+                this.removeDecision(classification);
+            }
+        }
+
+        if (this.decisions.has(decision.identifier)) {
+            // TODO: remove this once TS can recognize that we have check the
+            // key exists in the guard that uses Map.prototype.has
+            const currentValue = this.decisions.get(decision.identifier) as IDecision[];
+            this.decisions.set(decision.identifier, [...currentValue, decision]);
+        }
+
+        this.decisions.set(decision.identifier, [decision]);
+    }
```

Comment:
ok I can see you've moved the logic to the model, so one point. Model knows how to apply decisions.

But there's still so much logic here for what should be a simple map set/get. Are you sure the structure is right?